Tasks/VsTest/vstest.ts

```diff
@@ -61,35 +64,39 @@ export function startTest() {
             return;
         }
 
-        invokeVSTest(resultsDirectory)
-            .then(function (code) {
-                try {
-                    if (!isTiaAllowed()) {
-                        publishTestResults(resultsDirectory);
-                    }
-                    uploadVstestDiagFile();
-                    tl.setResult(code, tl.loc('VstestReturnCode', code));                    
-                } catch (error) {
-                    uploadVstestDiagFile();
-                    utils.Helper.publishEventToCi(AreaCodes.PUBLISHRESULTS, error.message, 1001, false);
-                    console.log('##vso[task.logissue type=error;code=' + error + ';TaskName=VSTest]');
-                    throw error; // This throw should be removed
+        invokeVSTest(resultsDirectory).then(function (code) {
+            try {
+                if (!isTiaAllowed()) {
+                    publishTestResults(resultsDirectory);
                 }
-            })
-            .fail(function (err) {
                 uploadVstestDiagFile();
-                utils.Helper.publishEventToCi(AreaCodes.INVOKEVSTEST, err.message, 1002, false);
-                console.log('##vso[task.logissue type=error;code=' + err + ';TaskName=VSTest]');
-                throw err;  // What is the point of this throw? Where is it getting caught?
-            });
+                if (code == tl.TaskResult.Failed) {
+                    tl.setResult(tl.TaskResult.Failed, tl.loc('VstestFailedReturnCode', code));
```

Comment:
Add a ci event here.

Reply:
Why do we need a ci event here? We are just setting the task result. Both error scenarios below have ci logging

Reply:
invokeVSTest(resultsDirectory).then(function (code)  
Based on the return value of this we are failing the task. A failure need not always be an exception. Something went wrong hence we are failing the task. To help identify the point at which we failed the task we need to have one here